### ISSUE TITLE
I seem to  find  a bug with dgl.batch() when I use GatedGraphConv for graph embedding

### ISSUE THREAD

**Issue by MAOCHEN-STU** [dmlc/dgl#3876]
## In order to accurately express the problem, please allow me to use Chinese to describe

在我使用GatedGraphConv进行图嵌入处理时，
        我在第一次时，只处理一个图数据，
        我在第二次时，使用 dgl.batch() 对50个图进行了批处理，然后使用 graph.batch_num_nodes()，按照索引值从GatedGraphConv批处理后的数据中切分出了 第一次处理时使用的图所对应的数据，
        发现两次所得的结果是不一样（我又随机选了一个数值在第一次和第二次的未切分结果中搜索，也没有找到相同的值）。
     （我设定了 torch.manual_seed 和 np.random.seed 来确保每次初始化的数据是相同的，同时图数据是同构的）
       （我是使用pip下载的 0.6.1 版本，在pycharm中，pytorch版本为 1.11.0）
       按照文档中对 dgl.batch() 中的描述，似乎在进行图嵌入时 批处理和单一处理的结果应该是相同的。
        我不确定是我的理解错误还是因为 dgl.batch() 引起的bug？或者有什么途径能够保证两种处理方式结果是一致的？

以下内容为两次数据的节选部分：

第一次的结果：
{"file_name": "1382_1", "label": 1, "graph_features": [[-0.007104884833097458, 0.01819194108247757, -0.051407475024461746, -0.02721775323152542, 0.04479151964187622,……

第二次的结果：
{"file_name": "1382_1", "label": 1, "graph_features": [[-0.040642425417900085, 0.030147546902298927, -0.07069188356399536, 0.05278344452381134, 0.03185143694281578, ……

以下为我使用的部分代码:

```
   def forward(self, batch_data, cuda=False, onedata=False):
  
        graph = batch_data
        features = batch_data.ndata['features']
        edge_types =batch_data.edata['etype']
        
        outputs = self.ggnn(graph, features, edge_types)

        if onedata:
            return outputs 

        embedding_result = []
        result_list = outputs.detach().numpy().tolist()
        num_list = graph.batch_num_nodes()

        for i in range(len(num_list)):
            if i == 0:
                start = 0
                end = num_list[0]
            embedding_result.append(result_list[start:end])
            start = end
            if i != (len(num_list) - 1):
                end = end + num_list[i + 1]

        return embedding_result
```

**Comment by Rhett-Ying** [comment]
Could you check whether the below demo I made simulates the scenario you described? It didn't hit the issue you mentioned.

Are you using the **same** `GatedGraphConv` instance for first single test and the second batch test? They should be the same or the initial model parameters are different.

```
import dgl
import numpy as np
import torch as th
from dgl.nn import GatedGraphConv

g = dgl.graph(([0,1,2,3,2,5], [1,2,3,4,0,3]))
feat = th.ones(6, 10)
conv = GatedGraphConv(10, 10, 2, 3)
etype = th.tensor([0,1,2,0,1,2])
res = conv(g, feat, etype)
print(res[0])

bg = dgl.batch([g,g])
et = th.cat((etype, etype))
ft = th.cat((feat, feat))
res2 = conv(bg, ft, et)
print(res2[0])

assert th.equal(res, res2[0:6])
```

**Comment by MAOCHEN-STU** [comment]
Hello, I copied 50 copies of the data of my first step to construct a batch according to your method, ran it again, and found that the result after two processing is the same. The problem seems to be in the code of my preprocessing data. The data of my second task adds an edge type due to batch processing, so the result is different from that after the first processing. Thanks for the reply, this led me to find a logic bug that was not easily detected.
